Add blog post about publishing Kubernetes SIG images to registry.k8s.io (#56246)
kubernetes/website: content/en/blog/_posts/2026/publishing-images-to-registry-k8s-io/index.md

---
layout: blog
title: "Publishing a Kubernetes SIG's Images to registry.k8s.io"
draft: true
slug: publishing-images-to-registry-k8s-io
author: >
  [Kahiro Okina](https://github.com/kahirokunn) (Craftsman Software, Inc.)
---

My first instinct was to use a release path I had seen in many GitHub-hosted
open source projects: build the image in GitHub Actions, push it to `ghcr.io`,
and publish from there.

Inside the `kubernetes-sigs` GitHub organization, that path stopped short. I
could build the images, but I could not point users at GHCR as an official
Kubernetes project distribution path. I learned in the `#github-management`
channel on the [Kubernetes Slack](https://slack.k8s.io/) that GHCR is not
supported for that role. Publishing the two images through the official route,
`registry.k8s.io`, ended up involving four repositories:

> `kubernetes-sigs/cluster-inventory-api`,

Reviewer: I think reading this directly in the introduction made me think `kubernetes-sigs/cluster-inventory-api` is part of the infrastructure setup though it gets cleared up later in the blog as this is an application repo, worth clarifying upfront so readers don't get confused from the start:
```suggestion
`<application-repository>` (here `kubernetes-sigs/cluster-inventory-api`),
```

Author: Thanks! I clarified this as the image-owning repository.

[`kubernetes/k8s.io`](https://github.com/kubernetes/k8s.io),
[`kubernetes/test-infra`](https://github.com/kubernetes/test-infra), and
[`kubernetes/org`](https://github.com/kubernetes/org).

The two images that finally shipped were:

```none
registry.k8s.io/cluster-inventory-api/secretreader:v0.1.3
registry.k8s.io/cluster-inventory-api/kubeconfig-secretreader:v0.1.3
```

The first successful `registry.k8s.io` release was `v0.1.2`; after I found a

> mistake, I re-released the images as `v0.1.3`.

Reviewer: I think the two image name code block _would fit better at the end of this section_ 
`"4. Promoting from staging to registry.k8s.io"`
cause I think putting _specific_ image names and tags in the intro before any context is established for what they are or why they exist  makes it harder for readers to follow

**_What we can do is replace it with this line here and move the code block to the end of section 4._** 

Lemme know what you think :)

```suggestion

The two plugin images for the `cluster-inventory-api` project ultimately shipped under `registry.k8s.io`.
#

Author: Thanks! I moved the concrete image names to the end of the promotion section and kept only a short summary in the intro.

project, but the procedure is not specific to that SIG.

## Why an image needed to be distributed at all

Cluster Inventory API is a SIG Multicluster project for helping applications and
tools discover, interact with, and make placement decisions across multiple
Kubernetes clusters. Its first major component is the `ClusterProfile` API, which
represents one cluster and the information consumers need to use it.

Some `ClusterProfile` entries point at access-provider plugins, such as
[`secretreader`](https://github.com/kubernetes-sigs/cluster-inventory-api/tree/main/plugins/secretreader/cmd/plugin),
that resolve how to reach a cluster. These plugins support being used as
[image volumes](/docs/tasks/configure-pod-container/image-volumes/).

The work started when Kueue's MultiKueue work asked for the `secretreader` plugin
to be provided as a reusable artifact
([cluster-inventory-api#34](https://github.com/kubernetes-sigs/cluster-inventory-api/issues/34)).

## Part 1: the road to publication

### 1. The original plan was ghcr.io

The first plan was to use a GitHub Actions workflow to build `plugins/*` with

> Docker Buildx on tag pushes and publish to GHCR (`ghcr.io`)

Reviewer: I think `build plugins/* with Docker Buildx` implies Docker is the only viable container build tool here so I guess we should make it  tool-agnostic:

```suggestion
The first plan was to use a GitHub Actions workflow to build container images for `plugins/*` on tag pushes and publish to GHCR (`ghcr.io`)
```

Author: Thanks, good point! I made the wording tool-agnostic and removed the Docker Buildx-specific phrasing.
